### ISSUE TITLE
MultiplayerSynchronizer is not able to sync some properties from some nodes (ie velocity from CharacterBody2D)

### ISSUE THREAD

**Issue by ricargoes** [godotengine/godot#68516]
### Godot version

4.0 beta4

### System information

Linux Manjaro - Vulkan API 1.2.0 - AMD Radeon RX 5700 (RADV NAVI10)

### Issue description

When trying to sync the CharacterBody2D built-in property `velocity` using the MultiplayerSynchronizer node, first it will not appear as an option in the helper wizard and if I try to specify its path directly it will not work.

![image](https://user-images.githubusercontent.com/8018280/201326879-e7656f78-d97f-420d-988e-af021618ce61.png)

I checked if this was the only missing property and it is not, for this node all the missing properties from the menu are:
```
max_slides
slide_on_ceiling
velocity
wall_min_slide_angle
```

I have checked this is also happening with other nodes like Camera2D

### Steps to reproduce

1. In any scene, create CharacterBody2D node
2. In the same scene create a MultiplayerSyncrhonizer node (it can be a child of the previous node)
3. Try to add the CharacterBody2D built-in property `velocity` to the replication menu, it will not appear.

### Minimal reproduction project

[DebugProject.zip](https://github.com/godotengine/godot/files/9989719/DebugProject.zip)

**Comment by ricargoes** [comment]
Ok, I have realized I can only synced exported properties. I do not know if this a feature or a bug since I cannot find any mention in the MultiplayerSynchronizer documentation.

In any case, I can imagine a variety of buit-in not-exported properties which could be interesting to synced; like the above mentioned, `CharacterBody2D`'s `velocity` or `Timer`'s `time_left`

**Comment by SamFlt** [comment]
Hi,

I'm also encountering this issue. 
Is this an expected behaviour that we should get around (by e.g syncing a custom property and update the velocity afterwards), or a current issue/bug?

Cheers!

**Comment by sygi** [comment]
It doesn't look like the expected behavior: in the official [multiplayer tutorial](https://godotengine.org/article/multiplayer-in-godot-4-0-scene-replication/), they are synchronizing this property.

**Comment by sygi** [comment]
Fwiw, I am using this workaround for now:
```
@export var my_velocity := Vector2(0., 0.):
    set(velo):
        velocity = velo  # to set mine
        my_velocity = velo  # to set theirs
```

and use `my_velocity` in the code. Note that this doesn't force updates to `my_velocity` when `velocity` changes (eg. if some underlying class does `velocity += 7`) as doing this requires a couple of lines to make sure the setters don't keep calling each other infinitely.

**Comment by Pspritechologist** [comment]
I can confirm that this issue is plaguing me in Godot version 4.1.1. I've used the same work around, but it's clearly less than ideal.

**Comment by zell2002** [comment]
same issue on 4.1.2 while following tutorial.
Oddly if you download the project and run it, it also cant find 'velocity' in that window, but it doesnt error during..
Where as in my project if i use the `Path:[ :property ]` input field, i can add it, but it casues errors

**Comment by eintopf** [comment]
Also an issue in 4.2 rc1 (using CharacterBody3D).

**Comment by Woor** [comment]
> Adding by path is possible: <img alt="Screenshot 2023-12-06 at 18 01 12" width="512" src="https://private-user-images.githubusercontent.com/100964/288367239-37d11a1b-e1f9-4ab5-8d5e-5c20e7d2e32c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNjA4OTksIm5iZiI6MTcwOTE2MDU5OSwicGF0aCI6Ii8xMDA5NjQvMjg4MzY3MjM5LTM3ZDExYTFiLWUxZjktNGFiNS04ZDVlLTVjMjBlN2QyZTMyYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyOFQyMjQ5NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ODAzYjJhMjRhMjdmN2M4N2M0MzMwMTYwY2ExMzg5ZTY0OTZhOTk3ZmY4YWI2ZTI3NmRmOGRiYmUxZGRkZTU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KT30psJgkLjPYiCtOlTIzCK9rJ42vD3HXrqAFbfHEPY">
> <img alt="Screenshot 2023-12-06 at 18 02 55" width="512" src="https://private-user-images.githubusercontent.com/100964/288367710-fc6f064d-b01b-4da7-87c3-133b916e10a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNjA4OTksIm5iZiI6MTcwOTE2MDU5OSwicGF0aCI6Ii8xMDA5NjQvMjg4MzY3NzEwLWZjNmYwNjRkLWIwMWItNGRhNy04N2MzLTEzM2I5MTZlMTBhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyOFQyMjQ5NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNGI1YjNjY2ZmYjExOTY2ZjVlNzM1ZmZiMjJkYTRlYjViOGFiYTFhNjNiZDA5MWRlZGFmYWE1OWFjYzQ1NDFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9vK29q8N5BT-ILAtPXCbQSfIoJ8uOnAeNyA7KAHBuFY">
> 
> But then: <img alt="Screenshot 2023-12-06 at 18 01 45" width="512" src="https://private-user-images.githubusercontent.com/100964/288367389-ec077c90-a4ce-433b-a4b9-cdd87051f53d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNjA4OTksIm5iZiI6MTcwOTE2MDU5OSwicGF0aCI6Ii8xMDA5NjQvMjg4MzY3Mzg5LWVjMDc3YzkwLWE0Y2UtNDMzYi1hNGI5LWNkZDg3MDUxZjUzZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyOFQyMjQ5NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzg2ZDE0ZmJmZTQ5YjFiYzFkYmMwMWY5ODVmMGE3NzMzN2Y4MmI0Y2Y4ZTJkMWFjNjU2MmIyZDBkOTc3MzE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sG64qNWcnnA0j-7yjpnk7h7EeDChXHqbq_i_QUcMwcc">

I've had this exact issue just now. 'velocity' is not in the list, and if you add it as 'Player:velocity' it will error.
However if you manually add it as just 'velocity' it will not and look the same in the Properties list.

**Comment by 3da** [comment]
Is it planned to fix?

**Comment by FireSquid6** [comment]
also encountering this issue on Godot 4.2.stable

**Comment by dmitrijbes** [comment]
As was already mentioned, the issue arises because of non-export properties not available through UI.
However, it's by design, because that's exactly what export annotation is used for.
It's not for a MultiplayerSynchronizer to decide, for other components, which properties to make available through UI.

I think that's exactly why addition via NodePath is available in the first place.
It's actually fully working, although a little bit confusing to use.
The issue with it appear, because people tend to specify NodePath from a scene root node.
While NodePath must be specified from MultiplayerSynchronizer root_path property.

That's why "player:velocity" doesn't work, because it's used directly as path.
While "velocity" works, because it's transformed into proper path ".:velocity".
You can see it clearly via inspecting source .tscn file.

In short, it's not a bug, but rather confusing UX, which should be specified in docs for MultiplayerSynchronizer.

![setup](https://github.com/godotengine/godot/assets/15231915/1ccbf04f-44b0-4ab0-9bfe-e80b679a7d2b)
![path](https://github.com/godotengine/godot/assets/15231915/0395cd6a-b570-4c3d-860b-bc6298b9ee0c)

**Comment by sygi** [comment]
Good that there is a reasonable workaround, but lack of docs in MultiplayerSynchronizer and ability to only add some of the properties via the UI and not others is a bug: there's no way for the user to know how exactly to use the synchronizer with some of the properties while the docs suggest it should work with all of them.

The documented semantics for `export` annotation doesn't mention relation to MultiplayerSynchronizer at all.

Either way, an ability to "export" a previously un-exported property sounds like a useful usability improvement.

**Comment by dmitrijbes** [comment]
One of the main functions of export annotation is to limit what kind of properties available to the user through the editor.
It's how it's designed and how it's used across all components. If component creator decided that property should not be exported, there is a specific reason for it and MultiplayerSynchronizer should not ignore or override it.
That's why MultiplayerSynchronizer respects export annotation, while giving other means to use non-exported properties.

MultiplayerSynchronizer does work with all properties, so it's not strictly a bug, but yeah, lack of properties in UI is confusing.
I think we both agree that it's at least non-obvious and requires clarification in docs.

**Comment by CrispinStichart** [comment]
In addition to updating the docs, it'd be nice if the [article about scene replication](https://godotengine.org/article/multiplayer-in-godot-4-0-scene-replication/) was updated as well, since it's still one of the top results when searching for information about Godot multiplayer.

**Comment by Correlander** [comment]
> As was already mentioned, the issue arises because of non-export properties not available through UI. However, it's by design, because that's exactly what export annotation is used for. It's not for a MultiplayerSynchronizer to decide, for other components, which properties to make available through UI.
> 
> I think that's exactly why addition via NodePath is available in the first place. It's actually fully working, although a little bit confusing to use. The issue with it appear, because people tend to specify NodePath from a scene root node. While NodePath must be specified from MultiplayerSynchronizer root_path property.
> 
> That's why "player:velocity" doesn't work, because it's used directly as path. While "velocity" works, because it's transformed into proper path ".:velocity". You can see it clearly via inspecting source .tscn file.
> 
> In short, it's not a bug, but rather confusing UX, which should be specified in docs for MultiplayerSynchronizer.
> 
> ![setup](https://private-user-images.githubusercontent.com/15231915/322873642-1ccbf04f-44b0-4ab0-9bfe-e80b679a7d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MDM3NDYsIm5iZiI6MTcxMzQwMzQ0NiwicGF0aCI6Ii8xNTIzMTkxNS8zMjI4NzM2NDItMWNjYmYwNGYtNDRiMC00YWIwLTliZmUtZTgwYjY3OWE3ZDJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDAxMjQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ0YWEzMjIwNmJlOGVmNGU5YjQ3MzY3Nzc2MjM5ZDQxYjFiNzY4MzRlNmJmYWI2MzQ1MmJmNmFiYTNmNTRmMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UfZTnmjgehogMFrTXd2PXE7zvXmS0CVIhQ7hQrnHiE0) ![path](https://private-user-images.githubusercontent.com/15231915/322873664-0395cd6a-b570-4c3d-860b-bc6298b9ee0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MDM3NDYsIm5iZiI6MTcxMzQwMzQ0NiwicGF0aCI6Ii8xNTIzMTkxNS8zMjI4NzM2NjQtMDM5NWNkNmEtYjU3MC00YzNkLTg2MGItYmM2Mjk4YjllZTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDAxMjQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3YmE5ZTM1ZDNjMWYwN2Y0OTY0ZWQ1OWM0YWEzNzU1YmQxYWY1NTNlNjU3ZGU3Y2E0N2FjZWQ5ZmI1NDgzZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OK3LHrDSErl-wv6SzbTXQaCtuVrpTX0oqMP4jVhQ_ts)

This doesn't seem to be the case for me, whether I type Player:velocity, :velocity, or just velocity when entering the path to the property to sync, it does not work. I have the position synced which I added through the editor, and also an exported variable, and they both sync properly. But velocity for some reason will not work, and I get errors saying the property doesn't exist. When opening the .tscn file directly, it is formatted exactly the same as these other two working properties (".:velocity")

**Comment by Shikamounty** [comment]
> As was already mentioned, the issue arises because of non-export properties not available through UI. However, it's by design, because that's exactly what export annotation is used for. It's not for a MultiplayerSynchronizer to decide, for other components, which properties to make available through UI.
> 
> I think that's exactly why addition via NodePath is available in the first place. It's actually fully working, although a little bit confusing to use. The issue with it appear, because people tend to specify NodePath from a scene root node. While NodePath must be specified from MultiplayerSynchronizer root_path property.
> 
> That's why "player:velocity" doesn't work, because it's used directly as path. While "velocity" works, because it's transformed into proper path ".:velocity". You can see it clearly via inspecting source .tscn file.
> 
> In short, it's not a bug, but rather confusing UX, which should be specified in docs for MultiplayerSynchronizer.
> 
> ![setup](https://private-user-images.githubusercontent.com/15231915/322873642-1ccbf04f-44b0-4ab0-9bfe-e80b679a7d2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDQ5NjMsIm5iZiI6MTcyMzE0NDY2MywicGF0aCI6Ii8xNTIzMTkxNS8zMjI4NzM2NDItMWNjYmYwNGYtNDRiMC00YWIwLTliZmUtZTgwYjY3OWE3ZDJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE5MTc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1NDgzMjYwNmJmZjAzZjY4NTk5NDk3ZGE4NWQ2NGYyYjZlNzE0YjIzOTg0YTc1MmNhZDgzOTdkMWFiMmRiMTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BpYewrJCGWRejX3ZKBDoP3NVLtXThIy7c2VurBi1gzY) ![path](https://private-user-images.githubusercontent.com/15231915/322873664-0395cd6a-b570-4c3d-860b-bc6298b9ee0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDQ5NjMsIm5iZiI6MTcyMzE0NDY2MywicGF0aCI6Ii8xNTIzMTkxNS8zMjI4NzM2NjQtMDM5NWNkNmEtYjU3MC00YzNkLTg2MGItYmM2Mjk4YjllZTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE5MTc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzMWNjZDg4MjdmNTZmNDQwMDk4NTk1ZTk0ZjEyMzU1N2QwNGViNTRhZDQzOTE1MmNiNzAyNmZmYmMwNzZiMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.r-IGpNYnGSfpmaaDPs_rlL9eydsfiOo_NFdgp4Dkclo)

Thanks a lot for the explanation
